app/models/target.rb

```diff
@@ -27,12 +29,20 @@ class Target < ActiveRecord::Base
     food: 8
   }
 
+  belongs_to :user
+
   validates :area_lenght, :title, presence: true
   validates :topic, presence: true, inclusion: { in: topics.keys }
   validates :latitude, presence: true,
                        numericality: { greater_than_or_equal_to: -90, less_than_or_equal_to: 90 }
   validates :longitude, presence: true,
                         numericality: { greater_than_or_equal_to: -180, less_than_or_equal_to: 180 }
+  validate :validate_target_limit, on: :create
 
-  belongs_to :user
+  private
+
+  def validate_target_limit
+    return errors.add(:targets, I18n.t('model.errors.to_many_targets')) unless
```

Comment:
I think could be clearer and easier to maintain the error as:
`model.errors.targets.to_many`

Reply:
Also, can you write `unless user&.targets.count < MAX_TARGETS_PER_USER` ?